src/a_plus_b.rs

```diff
@@ -9,10 +8,11 @@ fn main() {
                             .map(|i| i.parse::<i32>().ok())
                             .collect::<Vec<Option<i32>>>();
 
-    let sum = match &words[..] {
-        [Some(x), Some(y)] => x + y,
-            _ => panic!("Please enter 2 integers")
-    };
+    let sum = words.iter().fold(0, |a, &b| {
+            if let Some(x) =  b { a + x }
+            else { panic!("Please enter 2 integers") }
+        }
+    );
```

Comment:
I would prefer something more straightforward here... We just want to sum up two numbers. What do you think of the following?

```
let err_msg = "Please enter 2 integers";
let sum = if words.len() == 2 {
    words[0].expect(err_msg) + words[1].expect(err_msg)
} else {
    panic!(err_msg);
}
```